torch_geometric/explain/algorithm/pgm_explainer.py

```diff
@@ -0,0 +1,440 @@
+import logging
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+import torch
+from pgmpy.estimators.CITests import chi_square
+from torch import Tensor
+
+from torch_geometric.explain.config import ModelTaskLevel
+from torch_geometric.explain.explanation import Explanation
+
+from .base import ExplainerAlgorithm
+
+
+class PGMExplainer(ExplainerAlgorithm):
+    r"""The PGMExplainer model from the `"PGM-Explainer:
+     Probabilistic Graphical Model Explanations for Graph Neural Networks"
+    <https://arxiv.org/abs/2010.05788>
+
+
+    Args:
+
+        **kwargs (optional): Additional hyper-parameters to override default
+            settings in
+            :attr:`~torch_geometric.explain.algorithm.GNNExplainer.coeffs`.
+    """
+    def __init__(
+        self,
+        perturb_feature_list=None,
+        perturb_mode="mean",  # mean, zero, max or uniform
+        perturb_indicator="diff",  # diff or abs
+        **kwargs,
+    ):
+        super().__init__()
+        self.perturb_feature_list = perturb_feature_list
+        self.perturb_mode = perturb_mode
+        self.perturb_indicator = perturb_indicator
+        # edge_mask is only None if edge_mask_type is != MaskType.object
+
+        print("Explainer settings")
+        # print("Number of layers: ", self.num_layers)
+        print("Perturbation mode: ", self.perturb_mode)
+
+    def perturb_features_on_node(
+            self,
+            feature_matrix: torch.Tensor,
+            node_idx: int,
+            pertubation_mode="randint",
+            is_random: bool = False,
+            is_pertubation_scaled: bool = False,
+            perturb_feature_list=None  # indexes of features being pertubed
```

Comment:
It would be better to specify the types of `pertubation_mode` and `perturb_feature_list`